kratos/python/add_matrix_to_python.cpp

```diff
@@ -89,6 +89,25 @@ namespace Python
 
           return matrix;
         }));
+
+        matrix_binder.def(py::init([](const py::list& input){
+          std::size_t num_rows = input.size();
+          if( num_rows == 0 || ( (num_rows == 1) && ((py::cast<py::list>(input[0])).size() == 0) ) )
+            return DenseMatrix<double>(0,0);
+          else{
+            std::size_t num_cols = (py::cast<py::list>(input[0])).size();
+            DenseMatrix<double>matrix = DenseMatrix<double>(num_rows, num_cols);
+            for(std::size_t i = 0; i < num_rows; i++){
+              KRATOS_ERROR_IF( (py::cast<py::list>(input[i])).size() != num_cols ) << "Wrong size of a row\n";
```

Comment:
Is this check required ??? To make sure right number of columns in each row ...
Or should I remove this ?

Reply:
yes it is required

=> this should also be checked in the tests. I.e. you should add tests to make sure that the correct exceptions are thrown

Reply:
Test Added .